### ISSUE TITLE
Does it support Qwen series hosted model?

### ISSUE THREAD

**Issue by ruiqurm** [OpenInterpreter/open-interpreter#1572]
### Describe the solution you'd like

Hi. I want to use the Qwen-turbo model on the open interpreter. I checked the documentation and did not find the `Qwen` series hosted model. How can I add support for it? Is there a tutorial or guide to follow? 
Thanks.

**Comment by ebowwa** [comment]
Certainly! Let's dive straight into a concrete example of how you can integrate **DeepSeek V3** from **OpenRouter.ai** into your **Open Interpreter** setup. This will help you leverage DeepSeek's powerful language model capabilities within Open Interpreter seamlessly.

### **Step-by-Step Integration Guide**

#### **1. Obtain Your OpenRouter.ai API Key**

First, you need to sign up for an account on [[OpenRouter.ai](https://openrouter.ai/)](https://openrouter.ai/) and generate an API key.

1. **Sign Up / Log In:**
   - Go to [[OpenRouter.ai](https://openrouter.ai/)](https://openrouter.ai/) and create an account or log in if you already have one.

2. **Generate API Key:**
   - Navigate to the [[API Keys](https://openrouter.ai/account/api-keys)](https://openrouter.ai/account/api-keys) section in your account dashboard.
   - Click on **"Create New API Key"**.
   - Name your API key (e.g., `OpenInterpreter-DeepSeek`) and copy the generated key. **Keep it secure!**

#### **2. Install Open Interpreter**

If you haven't installed Open Interpreter yet, you can do so using `pip`:

```bash
pip install open-interpreter
```

#### **3. Configure Open Interpreter to Use DeepSeek V3 via OpenRouter.ai**

You have two primary ways to configure Open Interpreter to use DeepSeek V3: **Command Line** and **Python Script**.

---

##### **A. Using Command Line**

Open your terminal and run the following command, replacing `YOUR_OPENROUTER_API_KEY` with the API key you obtained:

```bash
interpreter --api_base "https://openrouter.ai/api/v1" --api_key "YOUR_OPENROUTER_API_KEY" --model "deepseek/deepseek-chat"
```

**Example:**

```bash
interpreter --api_base "https://openrouter.ai/api/v1" --api_key "sk-1234567890abcdef" --model "deepseek/deepseek-chat"
```

This command sets the API base to OpenRouter's endpoint, provides your API key, and specifies the DeepSeek V3 model.

---

##### **B. Using a Python Script**

If you prefer configuring within a Python environment, follow these steps:

1. **Create a Python Script (e.g., `use_deepseek.py`):**

```python
from interpreter import interpreter

# Configure the interpreter to use DeepSeek V3 via OpenRouter.ai
interpreter.llm.model = "deepseek/deepseek-chat"
interpreter.llm.api_key = "YOUR_OPENROUTER_API_KEY"
interpreter.llm.api_base = "https://openrouter.ai/api/v1"

# Start an interactive chat session
interpreter.chat()
```

2. **Replace the API Key:**

Ensure you replace `"YOUR_OPENROUTER_API_KEY"` with your actual API key.

3. **Run the Script:**

```bash
python use_deepseek.py
```

This script initializes Open Interpreter with DeepSeek V3 and starts an interactive chat session.

---

#### **4. Testing the Integration**

After configuring, it's essential to verify that the integration works as expected.

1. **Start Open Interpreter:**

   - **Command Line:** If you used the command line method, running the command as shown above will start the interpreter.
   - **Python Script:** Running your Python script will start the interpreter.

2. **Execute a Test Command:**

   Once the interpreter is running, try a simple command to ensure everything is functioning.

   **Example:**

   ```bash
   interpreter> Plot the stock prices of AAPL and MSFT for the last month.
   ```

   **Expected Outcome:**

   The interpreter should process the command using DeepSeek V3 and generate the corresponding plot.

---

#### **5. Advanced Configuration (Optional)**

You can further customize your setup based on your requirements.

- **Setting Environment Variables:**

  Instead of passing the API key directly in commands or scripts, you can set it as an environment variable for better security.

  ```bash
  export OPENROUTER_API_KEY="YOUR_OPENROUTER_API_KEY"
  interpreter --api_base "https://openrouter.ai/api/v1" --api_key "$OPENROUTER_API_KEY" --model "deepseek/deepseek-chat"
  ```

- **Using a `.env` File:**

  Create a `.env` file in your project directory:

  ```env
  OPENROUTER_API_KEY=your_api_key_here
  ```

  Then, modify your Python script to load environment variables:

  ```python
  import os
  from interpreter import interpreter
  from dotenv import load_dotenv

  load_dotenv()  # Load variables from .env

  interpreter.llm.model = "deepseek/deepseek-chat"
  interpreter.llm.api_key = os.getenv("OPENROUTER_API_KEY")
  interpreter.llm.api_base = "https://openrouter.ai/api/v1"

  interpreter.chat()
  ```

  **Install `python-dotenv` if not already installed:**

  ```bash
  pip install python-dotenv
  ```

---

#### **6. Troubleshooting Tips**

- **Invalid API Key:**
  - Ensure that the API key is correctly copied without any extra spaces or characters.
  
- **Network Issues:**
  - Verify your internet connection.
  - Ensure that `https://openrouter.ai/api/v1` is accessible from your network.

- **Model Not Found:**
  - Double-check the model name. It should be exactly `"deepseek/deepseek-chat"`.

- **Verbose Mode:**
  - Enable verbose mode to get detailed logs which can help in debugging.
  
  **Command Line:**

  ```bash
  interpreter --api_base "https://openrouter.ai/api/v1" --api_key "YOUR_OPENROUTER_API_KEY" --model "deepseek/deepseek-chat" --verbose
  ```

  **Python Script:**

  ```python
  interpreter.verbose = True
  ```

---

### **Conclusion**

By following the above steps, you should have successfully integrated **DeepSeek V3** via **OpenRouter.ai** into your **Open Interpreter** setup. This integration harnesses the advanced capabilities of DeepSeek, providing a more powerful and flexible language model experience.

If you encounter any further issues or have specific questions, feel free to reach out by commenting on [[Issue #1572](https://github.com/OpenInterpreter/open-interpreter/issues/1572)](https://github.com/OpenInterpreter/open-interpreter/issues/1572) or joining the [[Open Interpreter Discord](https://discord.com/invite/open-interpreter)](https://discord.com/invite/open-interpreter) for real-time support.

Happy coding!

**Comment by tristayunsub** [comment]
wow so we can use o1, deepseek whatever we want! thank you

**Comment by blakkd** [comment]
> ### Is your feature request related to a problem? Please describe.
> _No response_
> 
> ### Describe the solution you'd like
> Hi. I want to use the Qwen-turbo model on the open interpreter. I checked the documentation and did not find the `Qwen` series hosted model. How can I add support for it? Is there a tutorial or guide to follow? Thanks.
> 
> ### Describe alternatives you've considered
> _No response_
> 
> ### Additional context
> _No response_

Using the 1.0 preview, you can achieve what you look for using ollama: `interpreter --model ollama/qwen2.5-coder:32b-instruct-q4_K_M`.
No even need to pass `--api-base` or `--provider` litellm will figure out, syntax is that simple.

Just pay attention for now, only `ollama/[modelname]` is working, `ollama_chat/[modelname]` doesn't.

If anyone can help reporting the above chatgpt bullshit bot, so annoying...

**Comment by thedunston** [comment]
Also, openrouter isn't needed. DeepSeek has an OpenAI compatible AI:

interpreter --api_base "https://api.deepseek.com" --api_key "YOUR_KEY" --model "deepseek-chat"

**Comment by guxiangming** [comment]
> Also, openrouter isn't needed. DeepSeek has an OpenAI compatible AI:
> 
> interpreter --api_base "https://api.deepseek.com" --api_key "YOUR_KEY" --model "deepseek-chat"

如何持久化设置？不需要每次启动都声明

**Comment by PattonChen** [comment]
I just want to say * * * you. Your official website documents are really terrible.

![Image](https://github.com/user-attachments/assets/e0cdfaad-6986-495f-93c8-88833a753bd9)

Your documentation is very poor.  
IT really really waste my too much time .

**Comment by blakkd** [comment]
> I just want to say * * * you. Your official website documents are really terrible.
> 
> ![Image](https://github.com/user-attachments/assets/e0cdfaad-6986-495f-93c8-88833a753bd9)
> 
> Your documentation is very poor. IT really really waste my too much time .

@PattonChen Instead of insulting the work of others and making your little paint things, you could have saved time for everyone including yourself explaining what really is the issue...

**Comment by interpreterwork** [comment]
Thank you for reporting this and for helping build the classic Python version of Open Interpreter. This issue concerns the earlier Python/LiteLLM/Jupyter product; this repository now tracks the Rust Open Interpreter built on Codex. We are closing the legacy report here so current product tracking starts clean. If you still want to pursue it, the community-maintained classic fork is https://github.com/endolith/open-interpreter. If the same problem affects the current Interpreter release, please open a new issue with the current template and include the version, provider, model, wire API, harness, platform, and reproduction.
